pkg/agent/run.go

```diff
@@ -138,6 +146,37 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	return ctx.Err()
 }
 
+// getConntrackConfig uses the kube-proxy code to parse the user-provided extra-kube-proxy-args, and
+// extract the conntrack settings so that K3s can set them itself. This allows us to soft-fail when
+// running K3s in Docker, where kube-proxy is no longer allowed to set conntrack sysctls on newer kernels.
+func getConntrackConfig(nodeConfig *daemonconfig.Node) (*kubeproxyconfig.KubeProxyConntrackConfiguration, error) {
+	cmd := app2.NewProxyCommand()
+	cmd.ParseFlags(nodeConfig.AgentConfig.ExtraKubeProxyArgs)
+	maxPerCore, err := cmd.Flags().GetInt32("conntrack-max-per-core")
+	if err != nil {
+		return nil, err
+	}
+	min, err := cmd.Flags().GetInt32("conntrack-min")
+	if err != nil {
+		return nil, err
+	}
+	establishedTimeout, err := cmd.Flags().GetDuration("conntrack-tcp-timeout-established")
+	if err != nil {
+		return nil, err
+	}
+	closeWaitTimeout, err := cmd.Flags().GetDuration("conntrack-tcp-timeout-close-wait")
+	if err != nil {
+		return nil, err
+	}
+
+	return &kubeproxyconfig.KubeProxyConntrackConfiguration{
+		MaxPerCore:            utilpointer.Int32Ptr(maxPerCore),
+		Min:                   utilpointer.Int32Ptr(min),
+		TCPEstablishedTimeout: &metav1.Duration{Duration: establishedTimeout},
+		TCPCloseWaitTimeout:   &metav1.Duration{Duration: closeWaitTimeout},
+	}, nil
```

Comment:
These value should be `0` for rootless (https://github.com/kubernetes-sigs/kind/blob/515fb3aeb2ae4af6d8229754a152eccd016d1d15/pkg/cluster/internal/kubeadm/config.go#L422-L427), though k3s is patched to ignore the sysctl errors on rootless.

Reply:
Yeah, the idea is to attempt to set the default or user-provided values, but soft fail and just warn if they can't be set - and then set the kube-proxy args to zero so that kube-proxy itself doesn't bother trying to set them, as it will cause the whole process to exit if it can't.

Reply:
Changed these to 0 for rootless